packages/connection/src/class-error-handler.php

```diff
@@ -0,0 +1,568 @@
+ *
+ * Errors are stored in the database as options in the following format:
+ *
+ * [
+ *   $error_code => [
+ *     $user_id => [
+ *       $error_details
+ *     ]
+ *   ]
+ * ]
+ *
+ * For each error code we store a maximum of 5 errors for 5 different user ids.
+ *
+ * An user ID can be
+ * * 0 for blog tokens
+ * * positive integer for user tokens
+ * * 'invalid' for malformed tokens
+ *
+ * @since 8.7.0
+ */
+class Error_Handler {
+
+	/**
+	 * The name of the option that stores the errors
+	 *
+	 * @since 8.7.0
+	 *
+	 * @var string
+	 */
+	const STORED_ERRORS_OPTION = 'jetpack_connection_xmlrpc_errors';
+
+	/**
+	 * The name of the option that stores the errors
+	 *
+	 * @since 8.7.0
+	 *
+	 * @var string
+	 */
+	const STORED_VERIFIED_ERRORS_OPTION = 'jetpack_connection_xmlrpc_verified_errors';
+
+	/**
+	 * The prefix of the transient that controls the gate for each error code
+	 *
+	 * @since 8.7.0
+	 *
+	 * @var string
+	 */
+	const ERROR_REPORTING_GATE = 'jetpack_connection_error_reporting_gate_';
+
+	/**
+	 * Time in seconds a test should live in the database before being discarded
+	 *
+	 * @since 8.7.0
+	 */
+	const ERROR_LIFE_TIME = DAY_IN_SECONDS * 3;
+	/**
+	 * List of known errors. Only error codes in this list will be handled
+	 *
+	 * @since 8.7.0
+	 *
+	 * @var array
+	 */
+	public $known_errors = array(
+		'malformed_token',
+		'malformed_user_id',
+		'unknown_user',
+		'no_user_tokens',
+		'empty_master_user_option',
+		'no_token_for_user',
+		'token_malformed',
+		'user_id_mismatch',
+		'no_possible_tokens',
+		'no_valid_token',
+		'unknown_token',
+		'could_not_sign',
+		'invalid_scheme',
+		'invalid_secret',
+		'invalid_token',
+		'token_mismatch',
+		'invalid_body',
+		'invalid_signature',
+		'invalid_body_hash',
+		'invalid_nonce',
+		'signature_mismatch',
+	);
+
+	/**
+	 * Holds the instance of this singleton class
+	 *
+	 * @since 8.7.0
+	 *
+	 * @var Error_Handler $instance
+	 */
+	public static $instance = null;
+
+	/**
+	 * Initialize instance, hookds and load verified errors handlers
+	 *
+	 * @since 8.7.0
+	 */
+	private function __construct() {
+		defined( 'JETPACK__ERRORS_PUBLIC_KEY' ) || define( 'JETPACK__ERRORS_PUBLIC_KEY', 'KdZY80axKX+nWzfrOcizf0jqiFHnrWCl9X8yuaClKgM=' );
+
+		add_action( 'rest_api_init', array( $this, 'register_verify_error_endpoint' ) );
+
+		$this->handle_verified_errors();
+	}
+
+	/**
+	 * Gets the list of verified errors and act upon them
+	 *
+	 * @since 8.7.0
+	 *
+	 * @return void
+	 */
+	public function handle_verified_errors() {
+		$verified_errors = $this->get_verified_errors();
+		foreach ( $verified_errors as $error_code => $user_errors ) {
+
+			switch ( $error_code ) {
+				case 'malformed_token':
+				case 'token_malformed':
+				case 'no_possible_tokens':
+				case 'no_valid_token':
+				case 'unknown_token':
+				case 'could_not_sign':
+				case 'invalid_token':
+				case 'could_not_sign':
```

Comment:
This is a duplicate; there's a 'could_not_sign' case two lines above this one.

Reply:
thanks! fixed in https://github.com/Automattic/jetpack/pull/16194/commits/cbcf238b6cb6f0088f0cdf5e8c763b25ad65ad1e